### PR TITLE
[TRIVIAL] Fix NuDB build error

### PR DESCRIPTION
**Pull request by ximinez**
## High Level Overview of Change

Fix a build error that seems to be limited to non-unity builds on Windows (Visual Studio) due to a missing `#include` in NuDB.

### Context of Change

I think the reordering of `#includes` in the physical redesign PR (#4997) removed a silent dependency from one of the headers in NuDB. The dependency version of NuDB didn't change, as far as I can tell. CI didn't catch the issue because the Windows build uses unity to save time.

The error is:
```
C:\Users\passp\.conan\data\nudb\2.0.8\_\_\package\5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9\include\nudb/detail/stream.hpp(64,20): error C2039: 'logic_error': is not a member of 'std' [C:\Dev\rippled\upstream\build\cmake\msvc19.OFF\rippled.vcxproj]
```
`std::logic_error` requires `<stdexcept>`, but `nudb/detail/stream.hpp` does not include it.

My assumption is some other header that had previously been ordered before `stream.hpp` and included `<stdexcept>` is now ordered after `stream.hpp`. I don't think it's worth figuring out which file it is, because it's obviously a problem with NuDB.

### Type of Change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] Build

## Future Tasks

See also: https://github.com/cppalliance/NuDB/pull/100

### CODE REVIEW

**Comment by ximinez** [comment]
Rewrote this to patch the source via Conan.

**Review by thejohnfreeman — COMMENTED** [review]
Do you want to add an instruction to `BUILD.md` explaining how to export the right version of `nudb` from `external/nudb`?

File: BUILD.md [hunk 1]
```diff
@@ -98,6 +98,12 @@ Update the compiler settings:
    conan profile update settings.compiler.cppstd=20 default
    ```
 
+Configure Conan to use recipe revisions:
```
Comment (thejohnfreeman) [inline BUILD.md:101]:
```suggestion
Configure Conan (1.x only) to use recipe revisions:
```

**Review by thejohnfreeman — CHANGES_REQUESTED** [review]

File: BUILD.md (hunk 1 above)
Comment (ximinez) [reply BUILD.md]: Done

**Review by thejohnfreeman — APPROVED** [review]

**Pull request merged**